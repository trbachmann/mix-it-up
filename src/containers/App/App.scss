header {
  background-color: $tan;
}

h1 {
  font-family: 'Flamenco', cursive;
  padding-top: 0.7em;
  padding-left: .7em;
  padding-bottom: 0.7em;
  font-size: 3em;
  margin: 0;
}

body {
  background-color: $lightGray;
}

