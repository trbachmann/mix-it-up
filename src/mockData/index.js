export const mockRecipes = [
  {
    yield: "4 Servings",
    totalTime: "15 min",
    images: [
      {
        hostedSmallUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s90",
        hostedMediumUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s180",
        hostedLargeUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s360",
        imageUrlsBySize: {
          90: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s90-c",
          360: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s360-c"
        }
      }
    ],
    name: "Toasted Marshmallow Cookies and Cream Milkshake",
    source: {
      sourceDisplayName: "Picky Palate",
      sourceSiteUrl: "http://picky-palate.com",
      sourceRecipeUrl: "http://picky-palate.com/2013/02/15/toasted-marshmallow-cookies-and-cream-milkshake/"
    },
    id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
    ingredientLines: [
      "6 Jumbo marshmallows",
      "6 cups good quality vanilla ice cream",
      "1 cup milk",
      "6 Oreo Cookies"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047'>Toasted Marshmallow Cookies and Cream Milkshake recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047",
      text: "Toasted Marshmallow Cookies and Cream Milkshake recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 4,
    totalTimeInSeconds: 900,
    attributes: {
      course: [
        "Beverages",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    "flavors": {
      Piquant: 0,
      Meaty: 0.1667,
      Bitter: 0.1667,
      Sweet: 0.6667,
      Sour: 0.1667,
      Salty: 0.1667
    },
    rating: 5,
    notes: ''
  },
  {
    yield: "serves two",
    totalTime: "40 min",
    images: [
      {
        hostedSmallUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s90",
        hostedMediumUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s180",
        hostedLargeUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s360",
        imageUrlsBySize: {
          90: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s90-c",
          360: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s360-c"
        }
      }
    ],
    name: "The Best (and Easy) Peanut Butter Cookies",
    source: {
      "sourceDisplayName": "Julia's Album",
      "sourceSiteUrl": "juliasalbum.com",
      "sourceRecipeUrl": "https://juliasalbum.com/easy-peanut-butter-cookies/"
    },
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    ingredientLines: [
      "1/4 cup creamy peanut butter (can use natural one too)",
      "2 tablespoons unsalted butter , room temperature (not melted - just let the butter from refrigerator sit on the counter for about an hour or so)",
      "1/2 cup brown sugar",
      "1 egg",
      "1/2 teaspoon pure vanilla extract",
      "3/4 cup all-purpose flour , sifted or aerated (see important note below!)",
      "1/2 teaspoon baking powder",
      "2 tablespoons granulated sugar for coating"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930'>The Best (and Easy) Peanut Butter Cookies recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
      text: "The Best (and Easy) Peanut Butter Cookies recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 2,
    totalTimeInSeconds: 2400,
    attributes: {
      course: [
        "Afternoon Tea",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    flavors: {
      Piquant: 0,
      Meaty: 0.8333,
      Bitter: 0.1667,
      Sweet: 0.8333,
      Sour: 0,
      Salty: 0.3333
    },
    rating: 4,
    notes: ''
  }
];

export const mockRecipe = {
  yield: "serves 16-20",
  totalTime: "30 min",
  images: [
    {
      hostedSmallUrl: "https://lh3.googleusercontent.com/923lxP6iaPgt7L1ZbNQekJGjcoC6CYGMVjQZ_-NEsrw_V5rcKZLcNgyUuLv_jmRV9lkSN0mTbVz95ASXf4LawA=s90",
      hostedMediumUrl: "https://lh3.googleusercontent.com/923lxP6iaPgt7L1ZbNQekJGjcoC6CYGMVjQZ_-NEsrw_V5rcKZLcNgyUuLv_jmRV9lkSN0mTbVz95ASXf4LawA=s180",
      hostedLargeUrl: "https://lh3.googleusercontent.com/923lxP6iaPgt7L1ZbNQekJGjcoC6CYGMVjQZ_-NEsrw_V5rcKZLcNgyUuLv_jmRV9lkSN0mTbVz95ASXf4LawA=s360",
      imageUrlsBySize: {
        90: "https://lh3.googleusercontent.com/TvJfjV21XRXRSrS5qNcbtGRm7VGSTK2GY-bCAQnlodFjP2ZljyTeuUg7dmvR8VnhubINi4wKBQ10wKRGpPRXgDQ=s90-c",
        360: "https://lh3.googleusercontent.com/TvJfjV21XRXRSrS5qNcbtGRm7VGSTK2GY-bCAQnlodFjP2ZljyTeuUg7dmvR8VnhubINi4wKBQ10wKRGpPRXgDQ=s360-c"
      }
    }
  ],
  name: "Frozen Oreo Dessert",
  source: {
    sourceDisplayName: "Creations by Kara",
    sourceSiteUrl: "creationsbykara.com",
    sourceRecipeUrl: "https://www.creationsbykara.com/frozen-oreo-ice-cream-dessert/"
  },
  id: "Frozen-Oreo-Dessert-2621007",
  ingredientLines: [
    "1/4 cup butter, melted",
    "1 package of Oreos, finely crushed (I used my food processor)",
    "1/2 gallon of ice cream (Cookies & Cream & Chocolate Chip Mint are yummy)",
    "Hot fudge sauce"
  ],
  attribution: {
    html: "<a href='http://www.yummly.co/recipe/Frozen-Oreo-Dessert-2621007'>Frozen Oreo Dessert recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
    url: "http://www.yummly.co/recipe/Frozen-Oreo-Dessert-2621007",
    "text": "Frozen Oreo Dessert recipes: information powered by Yummly",
    logo: "https://static.yummly.co/api-logo.png"
  },
  numberOfServings: 18,
  totalTimeInSeconds: 1800,
  attributes: {
    course: [
      "Desserts"
    ]
  },
  flavors: {},
  rating: 4,
  notes: ''
};

export const mockUserRecipeFromStorage = {
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    notes: 'Increase peanut butter by 1/2 cup and increase vanilla 2 teaspoons'
};

export const mockUserRecipeToChangeNotes = {
  id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
  notes: 'Increase peanut butter by 1/2 cup and increase vanilla 2 teaspoons'
};

export const mockRecipesWithUserNotes = [
  {
    yield: "4 Servings",
    totalTime: "15 min",
    images: [
      {
        hostedSmallUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s90",
        hostedMediumUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s180",
        hostedLargeUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s360",
        imageUrlsBySize: {
          90: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s90-c",
          360: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s360-c"
        }
      }
    ],
    name: "Toasted Marshmallow Cookies and Cream Milkshake",
    source: {
      sourceDisplayName: "Picky Palate",
      sourceSiteUrl: "http://picky-palate.com",
      sourceRecipeUrl: "http://picky-palate.com/2013/02/15/toasted-marshmallow-cookies-and-cream-milkshake/"
    },
    id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
    ingredientLines: [
      "6 Jumbo marshmallows",
      "6 cups good quality vanilla ice cream",
      "1 cup milk",
      "6 Oreo Cookies"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047'>Toasted Marshmallow Cookies and Cream Milkshake recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047",
      text: "Toasted Marshmallow Cookies and Cream Milkshake recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 4,
    totalTimeInSeconds: 900,
    attributes: {
      course: [
        "Beverages",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    "flavors": {
      Piquant: 0,
      Meaty: 0.1667,
      Bitter: 0.1667,
      Sweet: 0.6667,
      Sour: 0.1667,
      Salty: 0.1667
    },
    rating: 5,
    notes: ''
  },
  {
    yield: "serves two",
    totalTime: "40 min",
    images: [
      {
        hostedSmallUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s90",
        hostedMediumUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s180",
        hostedLargeUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s360",
        imageUrlsBySize: {
          90: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s90-c",
          360: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s360-c"
        }
      }
    ],
    name: "The Best (and Easy) Peanut Butter Cookies",
    source: {
      "sourceDisplayName": "Julia's Album",
      "sourceSiteUrl": "juliasalbum.com",
      "sourceRecipeUrl": "https://juliasalbum.com/easy-peanut-butter-cookies/"
    },
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    ingredientLines: [
      "1/4 cup creamy peanut butter (can use natural one too)",
      "2 tablespoons unsalted butter , room temperature (not melted - just let the butter from refrigerator sit on the counter for about an hour or so)",
      "1/2 cup brown sugar",
      "1 egg",
      "1/2 teaspoon pure vanilla extract",
      "3/4 cup all-purpose flour , sifted or aerated (see important note below!)",
      "1/2 teaspoon baking powder",
      "2 tablespoons granulated sugar for coating"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930'>The Best (and Easy) Peanut Butter Cookies recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
      text: "The Best (and Easy) Peanut Butter Cookies recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 2,
    totalTimeInSeconds: 2400,
    attributes: {
      course: [
        "Afternoon Tea",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    flavors: {
      Piquant: 0,
      Meaty: 0.8333,
      Bitter: 0.1667,
      Sweet: 0.8333,
      Sour: 0,
      Salty: 0.3333
    },
    rating: 4,
    notes: 'Increase peanut butter by 1/2 cup and increase vanilla 2 teaspoons'
  }
];

export const mockAttribution = { 
  attribution: {
  html: "Recipe search powered by <a href='http://www.yummly.co/recipes'><img alt='Yummly' src='https://static.yummly.co/api-logo.png'/></a>",
  url: "http://www.yummly.co/recipes/",
  text: "Recipe search powered by Yummly",
  logo: "https://static.yummly.co/api-logo.png"
    }
}

export const mockRecipesAfterNoteDelete = [
  {
    yield: "4 Servings",
    totalTime: "15 min",
    images: [
      {
        hostedSmallUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s90",
        hostedMediumUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s180",
        hostedLargeUrl: "http://lh4.ggpht.com/eUAHA16-zMjrkoD1qSyD-5GVyckd__2WwY213f-rIi7NfZbCmwfQ53pN9COxZ6q9VZBTbc0weJvJhy8NoVymLQ=s360",
        imageUrlsBySize: {
          90: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s90-c",
          360: "http://lh5.ggpht.com/BtZ6SmVreIVczKceXBa6Chq79PajwSVK047clqtMnSCQ5nlnnfhCEhSl6xzwDoovGG1eVzUnH1NUoHgXazCH=s360-c"
        }
      }
    ],
    name: "Toasted Marshmallow Cookies and Cream Milkshake",
    source: {
      sourceDisplayName: "Picky Palate",
      sourceSiteUrl: "http://picky-palate.com",
      sourceRecipeUrl: "http://picky-palate.com/2013/02/15/toasted-marshmallow-cookies-and-cream-milkshake/"
    },
    id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
    ingredientLines: [
      "6 Jumbo marshmallows",
      "6 cups good quality vanilla ice cream",
      "1 cup milk",
      "6 Oreo Cookies"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047'>Toasted Marshmallow Cookies and Cream Milkshake recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/Toasted-marshmallow-cookies-and-cream-milkshake-335047",
      text: "Toasted Marshmallow Cookies and Cream Milkshake recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 4,
    totalTimeInSeconds: 900,
    attributes: {
      course: [
        "Beverages",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    "flavors": {
      Piquant: 0,
      Meaty: 0.1667,
      Bitter: 0.1667,
      Sweet: 0.6667,
      Sour: 0.1667,
      Salty: 0.1667
    },
    rating: 5,
    notes: ''
  },
  {
    yield: "serves two",
    totalTime: "40 min",
    images: [
      {
        hostedSmallUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s90",
        hostedMediumUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s180",
        hostedLargeUrl: "https://lh3.googleusercontent.com/IL4SAQ143siXT-QfTZjeNORYdW5BTxDXLdQw-Ix1-rBMLlTfHKhZZFajd3myiwdIxndI9Kv3mpdzetFJPz7qfQ=s360",
        imageUrlsBySize: {
          90: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s90-c",
          360: "https://lh3.googleusercontent.com/JwhZ5F6bm4I67nr2WFSuv_anMW33tuBcWk1n7zzu6zIaAJraUPoogaJexrDUKsIQ2nrEWwtmxlFGSr533anG=s360-c"
        }
      }
    ],
    name: "The Best (and Easy) Peanut Butter Cookies",
    source: {
      "sourceDisplayName": "Julia's Album",
      "sourceSiteUrl": "juliasalbum.com",
      "sourceRecipeUrl": "https://juliasalbum.com/easy-peanut-butter-cookies/"
    },
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    ingredientLines: [
      "1/4 cup creamy peanut butter (can use natural one too)",
      "2 tablespoons unsalted butter , room temperature (not melted - just let the butter from refrigerator sit on the counter for about an hour or so)",
      "1/2 cup brown sugar",
      "1 egg",
      "1/2 teaspoon pure vanilla extract",
      "3/4 cup all-purpose flour , sifted or aerated (see important note below!)",
      "1/2 teaspoon baking powder",
      "2 tablespoons granulated sugar for coating"
    ],
    attribution: {
      html: "<a href='http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930'>The Best (and Easy) Peanut Butter Cookies recipe</a> information powered by <img alt='Yummly' src='https://static.yummly.co/api-logo.png'/>",
      url: "http://www.yummly.co/recipe/The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
      text: "The Best (and Easy) Peanut Butter Cookies recipes: information powered by Yummly",
      logo: "https://static.yummly.co/api-logo.png"
    },
    numberOfServings: 2,
    totalTimeInSeconds: 2400,
    attributes: {
      course: [
        "Afternoon Tea",
        "Desserts"
      ],
      cuisine: [
        "Kid-Friendly"
      ]
    },
    flavors: {
      Piquant: 0,
      Meaty: 0.8333,
      Bitter: 0.1667,
      Sweet: 0.8333,
      Sour: 0,
      Salty: 0.3333
    },
    rating: 4,
    notes: ''
  }
];

export const mockYummlyResponse = {
  "criteria": {
    "q": "desserts",
    "requirePictures": true,
    "allowedIngredient": null,
    "excludedIngredient": null
  },
  "matches": [
    {
      sourceDisplayName: "Picky Palate",
      ingredientLines: [
        "6 Jumbo marshmallows",
        "6 cups good quality vanilla ice cream",
        "1 cup milk",
        "6 Oreo Cookies"
      ],
      id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
      recipeName: "Toasted Marshmallow Cookies and Cream Milkshake",
      attributes: {
        course: [
          "Desserts"
        ]
      }
    }
  ],
  "facetCounts": {},
  "totalMatchCount": 353293,
  "attribution": {
    "html": "Recipe search powered by <a href='http://www.yummly.co/recipes'><img alt='Yummly' src='https://static.yummly.co/api-logo.png'/></a>",
    "url": "http://www.yummly.co/recipes/",
    "text": "Recipe search powered by Yummly",
    "logo": "https://static.yummly.co/api-logo.png"
  }
};

export const mockResultFromCombineMatches = [
  {
    name: "Toasted Marshmallow Cookies and Cream Milkshake",
    sourceDisplayName: "Picky Palate",
    id: "Toasted-marshmallow-cookies-and-cream-milkshake-335047",
    ingredientLines: [
      "6 Jumbo marshmallows",
      "6 cups good quality vanilla ice cream",
      "1 cup milk",
      "6 Oreo Cookies"
    ]
  },
  {
    id: "The-Best-_and-Easy_-Peanut-Butter-Cookies-2665930",
    notes: 'Increase peanut butter by 1/2 cup and increase vanilla 2 teaspoons'
  }
];