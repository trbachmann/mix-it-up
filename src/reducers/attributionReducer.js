export const attributionReducer = (state = {}, action) => {
  switch(action.type) {
    case 'SET_ATTRIBUTION':
      return action.attribution;
    default:
      return state;
  }
}