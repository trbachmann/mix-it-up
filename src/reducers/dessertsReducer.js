export const dessertsReducer = (state = [], action) => {
  switch(action.type) {
    case 'SET_RECIPES':
      return action.recipes;
    default:
      return state;
  }
}