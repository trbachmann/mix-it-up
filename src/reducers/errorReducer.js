export const errorReducer = (state = '', action) => {
  switch(action.type) {
    default:
      return state;
  }
}