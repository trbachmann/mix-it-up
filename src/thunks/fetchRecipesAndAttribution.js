import { fetchData } from '../utils/fetchData';
import { toggleLoading, setRecipes, setAttribution, setError } from '../actions';
import { fetchRecipes } from './fetchRecipes';
import { combineMatchesAndUserRecipes } from '../utils/combineMatchesAndUserRecipes';

export const fetchRecipesAndAttribution = (userRecipes) => {
  return async (dispatch) => {
    try {
      dispatch(toggleLoading(true));
      const url = 'https://api.yummly.com/v1/api/recipes?_app_id=' + process.env.REACT_APP_API_ID + '&_app_key=' + process.env.REACT_APP_API_KEY + '&q=desserts&requirePictures=true&maxResult=32&start=40';
      const response = await fetchData(url);
      const recipesToFetch = await combineMatchesAndUserRecipes(response.matches, userRecipes);
      const recipes = await dispatch(fetchRecipes(recipesToFetch));
      const updateRecipes = recipes.filter(
        recipe =>
          recipe.id !==
          'No-Bake-Keto-Peanut-Butter-Chocolate-Bars-2537418'
      );
      dispatch(toggleLoading(false));
      dispatch(setRecipes(updateRecipes));
      dispatch(setAttribution(response.attribution));
    } catch(error) {
      dispatch(setError(error.message));
    }
  }
}