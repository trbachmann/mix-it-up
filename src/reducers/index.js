import { combineReducers } from 'redux';
import { dessertsReducer } from './dessertsReducer';
import { userRecipesReducer } from './userRecipesReducer';
import { errorReducer } from './errorReducer';
import { statusReducer } from './statusReducer';
import { isLoadingReducer } from './isLoadingReducer';

export const rootReducer = combineReducers({
  desserts: dessertsReducer,
  userRecipes: userRecipesReducer,
  error: errorReducer,
  status: statusReducer,
  isLoading: isLoadingReducer
});