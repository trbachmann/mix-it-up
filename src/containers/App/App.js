import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { fetchMatches } from '../../thunks/fetchMatches';

export class App extends Component {

  componentDidMount() {
    this.props.fetchMatches();
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Mix It Up</h1>
        </header>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  desserts: state.desserts,
  error: state.error,
  isLoading: state.isLoading,
  attribution: state.attribution
});

export const mapDispatchToProps = (dispatch) => ({
  fetchMatches: () => dispatch(fetchMatches())
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
