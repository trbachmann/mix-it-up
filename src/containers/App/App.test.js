import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { createStore } from 'redux';
import { rootReducer } from '../../reducers/';
import { shallow } from 'enzyme';
import * as data from '../../mockData';
import { fetchRecipesAndAttribution } from '../../thunks/fetchRecipesAndAttribution';

const mockProps = {
  desserts: data.mockRecipes,
  userRecipes: [],
  error: '',
  isLoading: false,
  attribution: {},
  fetchRecipesAndAttribution: jest.fn(() => true),
}

jest.mock('../../thunks/fetchRecipesAndAttribution.js');

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const store = createStore(rootReducer);
    ReactDOM.render(<Provider store={store}><BrowserRouter><App {...mockProps}/></BrowserRouter></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  describe('App component', () => {
    let wrapper;
    
    beforeEach(() => {
      wrapper = shallow(<App {...mockProps}/>)
    })
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    afterEach(() => {
      localStorage.removeItem('userRecipes');
    });

    it('should match the snapshot when loading', () => {
      wrapper = shallow(<App {...mockProps} isLoading={true}/>);
      expect(wrapper).toMatchSnapshot();
    });

    describe('componentDidMount', () => {
      it('should call fetchRecipesAndAttribution if nothing in localStorage', () => {
        wrapper.instance().componentDidMount();
        expect(mockProps.fetchRecipesAndAttribution).toHaveBeenCalled();
      });

      it('should call fetchRecipesAndAttribution with savedUserRecipes', () => {
        const expected = data.mockUserRecipeFromStorage
        localStorage.setItem('userRecipes', JSON.stringify(data.mockUserRecipeFromStorage));
        wrapper.instance().componentDidMount();
        expect(mockProps.fetchRecipesAndAttribution).toHaveBeenCalledWith(expected);
      });
    });

    describe('calculateTotalUserRecipes', () => {
      it('should calculate the correct number of user recipes', () => {
        const expected = 1;
        wrapper = shallow(<App {...mockProps} desserts={data.mockRecipesWithUserNotes}/>)
        const result = wrapper.instance().calculateTotalUserRecipes();
        expect(result).toEqual(expected);
      });
    });

    describe('getRecipeRoute', () => {
      it('should return an Error404 if there is no match', () => {
        const mockPath = { params: { id: 'mockRecipe'} };
        const result = wrapper.instance().getRecipeRoute({ match: mockPath });
        const error404Wrapper = shallow(result);
        expect(error404Wrapper.find('.Error404')).toHaveLength(1);
      });

    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with desserts, error, isLoadind and attribution', () => {
      const mockState = {
        desserts: data.mockRecipes,
        error: '',
        status: '',
        isLoading: false,
        attribution: data.mockAttribution,
      };
      const expected = {
        desserts: data.mockRecipes,
        error: '',
        isLoading: false,
        attribution: data.mockAttribution,
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with a fetchRecipesAndAttribution', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = fetchRecipesAndAttribution([ data.mockUserRecipeFromStorage ]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.fetchRecipesAndAttribution([ data.mockUserRecipeFromStorage ]);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
