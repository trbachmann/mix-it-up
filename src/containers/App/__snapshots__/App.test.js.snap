// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component should match the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App--header"
  >
    <h1
      className="App--h1"
    >
      Mix It Up
    </h1>
    <Nav
      totalUserRecipes={0}
    />
  </header>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/desserts"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/my-recipes"
    />
    <Route
      path="/my-recipes/:id"
      render={[Function]}
    />
    <Route
      path="/desserts/:id"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <div
    className="App--div-attribution"
  >
    <p>
      Recipe search powered by
      <a
        className="App--div-anchor"
      >
        <img
          alt="Yummly"
        />
      </a>
    </p>
  </div>
</div>
`;

exports[`App App component should match the snapshot when loading 1`] = `
<div
  className="App"
>
  <header
    className="App--header"
  >
    <h1
      className="App--h1"
    >
      Mix It Up
    </h1>
    <Nav
      totalUserRecipes={0}
    />
  </header>
  <Loading />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/desserts"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/my-recipes"
    />
    <Route
      path="/my-recipes/:id"
      render={[Function]}
    />
    <Route
      path="/desserts/:id"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <div
    className="App--div-attribution"
  >
    <p>
      Recipe search powered by
      <a
        className="App--div-anchor"
      >
        <img
          alt="Yummly"
        />
      </a>
    </p>
  </div>
</div>
`;
